src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -85,7 +83,7 @@ def get_data_batch(self, train, batch_size, seed=0):
         train_ix = list(train_ix) + list(train_ix)  # duplicate to cover ranges past the end of the set
         return train[train_ix[start_i: stop_i]]
 
-    def train(self, data: Union[DataFrame, array], label:str, train_arguments: TrainParameters, num_cols: List[str],
+    def train(self, data: Union[DataFrame, array], label_col: str, train_arguments: TrainParameters, num_cols: List[str],
               cat_cols: List[str], preprocess: bool = True):
```

Comment:
Assumption 1: All data is passed in the data argument, the condition columns are included in this object
Assumption 2: A single column from the original data is expected to be identified by the label_col str. Not a priority but eventually this should be easily extendable for a list of conditions p.e.